content/python/concepts/deque/deque.md

```diff
@@ -0,0 +1,68 @@
+---
+Title: 'deque()'
+Description: 'A deque data structure in Python.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Queues'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+
+---
+
+The `deque`, short for _double-ended queue_," is a [Python](https://www.codecademy.com/resources/docs/python) data structure that allows for the efficient addition and removal of elements from both ends. It is a component of the [collections module](https://www.codecademy.com/resources/docs/python/collections-module) and serves as an alternative to the list where frequent insertions and deletions occur at both ends. Deques are notably advantageous when a queue is needed to enable fast appends and pops from both ends, or when a stack is required to support the same operations efficiently.
+
+## Syntax
+
+Initialize a deque with optional parameters `iterable` and `maxlen`.
+
+```pseudo
+d = deque(iterable=[], maxlen=None)
+
+In this syntax, deque() creates a new deque object, which can be initialized with an optional iterable and a maximum length. The append() and appendleft() methods add items to the right and left ends, respectively.
+The pop() and popleft() methods remove items from the corresponding ends. The rotate() method moves elements from one end to the other, effectively rotating the deque.
+
+## Example
+
+```py
+from collections import deque
+
+# Create a deque variable
+numbers_deque = deque([1, 2, 3, 4, 5])
+
+# Append a number to the right end
+numbers_deque.append(6)
+
+# Append a number to the left end
+numbers_deque.appendleft(0)
+
+# Pop a number from the right end
+numbers_deque.pop()
+
+# Pop a number from the left end
+numbers_deque.popleft()
+
+# Print the deque
+print(numbers_deque)
```

Comment:
In this case, instead of using the `.append()`, `.appendleft()` and other methods, lets stick to showing different use cases of deque itself, you can change the iterables, example:

#Use different iterables to create deques
from collections import deque
a = deque((8,7,9,6))
print(a)

b = deque([45, 845, 65])
print(b)

c= deque(range(5,10))
print(c)

d = deque("wxyz")
print(d)

numbers = {"firstname": "any name", "lastname": "any", "age": 25}
e = deque(numbers.keys())
print(e)

f = deque(numbers.values())
print(f)

g = deque(numbers.items())
print(g)

These are just some examples, you can create your own.